src/graph.py

```diff
@@ -33,19 +37,17 @@ def add_node(self, n):
         if n in self.graph:
             raise ValueError('Node already exists please try again')
         else:
-            self.graph[n] = []
+            self.graph[n] = {}
 
-    def add_edge(self, n1, n2):
+    def add_edge(self, n1, n2, w):
         '''adds a new edge to the graph connecting n1 and n2, if either
            n1 or n2 are not already present in the graph, they are added.'''
         if n2 not in self.graph:
-            self.graph[n2] = []
-        if n1 in self.graph:
-            self.graph[n1].append(n2)
-        else:
-            self.graph[n1] = [n2]
+            self.graph[n2] = {}
+        self.graph[n1] = {n2: w}
```

Comment:
Uh-oh, this ain't gonna fly. What happens if you want n1 to have an edge to more than one node? Your tests should be covering this possiblity?